service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/CodeableConceptCdMapper.java

```diff
@@ -87,6 +87,43 @@ public String mapCodeableConceptToCd(CodeableConcept codeableConcept) {
         return TemplateUtils.fillTemplate(CODEABLE_CONCEPT_CD_TEMPLATE, builder.build());
     }
 
+    // Medications are currently using D&T Codes rather than snomed codes but are being passed through as SNOMED codes which is
+    // creating a degradation on the receiving side. Until the types are configured correctly and agreed to a specification
+    // we have agreed to use the Concept ID rather than Description Id for medications which will avoided the degradation.
+    public String mapCodeableConceptForMedication(CodeableConcept codeableConcept) {
+        var builder = CodeableConceptCdTemplateParameters.builder();
+        var mainCode = findMainCode(codeableConcept);
+
+        builder.nullFlavor(mainCode.isEmpty());
+
+        if (mainCode.isPresent()) {
+            var extension = retrieveDescriptionExtension(mainCode.get())
+                .map(Extension::getExtension)
+                .orElse(Collections.emptyList());
+
+            builder.mainCodeSystem(SNOMED_SYSTEM_CODE);
+
+            Optional<String> code = Optional.of(mainCode.get().getCode().toString());
+            code.ifPresent(builder::mainCode);
+
+            Optional<String> displayName = extension.stream()
+                .filter(displayExtension -> DESCRIPTION_DISPLAY.equals(displayExtension.getUrl()))
+                .map(description -> description.getValue().toString())
+                .findFirst()
+                .or(() -> Optional.of(mainCode.get().getDisplay()));
```

Comment:
Same here, could `getDisplay()` return null?  `Optional.of` can throw a `NullPointerException`